analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -1637,3 +1637,33 @@ public void BoolSymbols_FalseFalse()
         public void FailsWhenFalseAny([DoesNotReturnIf(false)] bool condition1, [DoesNotReturnIf(false)] bool condition2) { }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/6176
+public class Repro_6176
+{
+    public void ExpressionArguments(object arg)
+    {
+        Select(x => x.FirstOrDefault().ToString());         // Compliant - custom expression processor, like Entity Framework, can propagate the "null" without executing the actual code
+        Select(() => arg == null ? arg.ToString() : null);  // Compliant, we don't analyze arguments of expression
+
+        System.Linq.Expressions.Expression<Func<IEnumerable<object>, object>> expr;
+        expr = x => x.FirstOrDefault().ToString();          // Compliant as well
+    }
+
+    public void DelegateArguments(object arg)
+    {
+        Invoke(x => x.FirstOrDefault().ToString());         // Noncompliant
+        Invoke(() => arg == null ? arg.ToString() : null);  // Noncompliant
+    }
+
+    public void Nested(object arg)
+    {
+        // To fix this, we'd need to bump referenced Roslyn version or shim SyntaxNodeExtensions.IsInExpressionTree for C# and VB, and replace SymbolicExecutionRunnerBase.IsInLinqExpression with that
```

Comment:
The `SyntaxNodeExtensions` classes in question are internal. Bumping the compiler version or shimming does not work. We need to copy the code.
```suggestion
        // To fix this, we'd need to copy SyntaxNodeExtensions.IsInExpressionTree for C# and VB, and replace SymbolicExecutionRunnerBase.IsInLinqExpression with that
```

Reply:
Removed